contracts/zero-ex/contracts/src/external/AllowanceTarget.sol

```diff
@@ -46,10 +46,23 @@ contract AllowanceTarget is
         onlyAuthorized
         returns (bytes memory resultData)
```

Comment:
I reckon we could shave off some more gas by putting in the fallback and doing the `onlyAuthorized` check in assembly. Here's [sample code](https://github.com/0xProject/0x-monorepo/blob/development/contracts/asset-proxy/contracts/src/ERC20Proxy.sol#L52) from the `ERC20Proxy`.

Reply:
I will give that a try. Thanks.

Reply:
This is done in the latest commit. It saved an additional 256 gas. Thanks!